docs/proposals/fleetapiserver/fleetapiserver.md

```diff
@@ -0,0 +1,239 @@
+| update           | PUT /apis/rbac.authorization.k8s.io/v1/clusterroles/name    | foo (specific cluster name) | replace the specified ClusterRole in cluster `foo`                                                                              |
+| update           | PUT /apis/rbac.authorization.k8s.io/v1/clusterroles/name    | karmada                     | replace the specified ClusterRole in Karmada control plane                                                                      |
+| update           | PUT /apis/rbac.authorization.k8s.io/v1/clusterroles/name    | all                         | replace the specified ClusterRole in cluster `yy` with input parameter name that meets the format `xx.clusterspace.yy`          |
+| patch            | PATCH /apis/rbac.authorization.k8s.io/v1/clusterroles/name  | foo (specific cluster name) | partially update the specified ClusterRole in cluster `foo`                                                                     |
+| patch            | PATCH /apis/rbac.authorization.k8s.io/v1/clusterroles/name  | karmada                     | partially update the specified ClusterRole in Karmada control plane                                                             |
+| patch            | PATCH /apis/rbac.authorization.k8s.io/v1/clusterroles/name  | all                         | partially update the specified ClusterRole in cluster `yy` with input parameter name that meets the format `xx.clusterspace.yy` |
+| delete           | DELETE /apis/rbac.authorization.k8s.io/v1/clusterroles/name | foo (specific cluster name) | delete a ClusterRole in cluster `foo`                                                                                           |
+| delete           | DELETE /apis/rbac.authorization.k8s.io/v1/clusterroles/name | karmada                     | delete a ClusterRole in Karmada control plane                                                                                   |
+| delete           | DELETE /apis/rbac.authorization.k8s.io/v1/clusterroles/name | all                         | delete a ClusterRole in cluster `yy` with input parameter name that meets the format `xx.clusterspace.yy`                       |
+| deletecollection | DELETE /apis/rbac.authorization.k8s.io/v1/clusterroles      | foo (specific cluster name) | delete collection of ClusterRole in cluster `foo`                                                                               |
+| deletecollection | DELETE /apis/rbac.authorization.k8s.io/v1/clusterroles      | karmada                     | delete collection of ClusterRole in Karmada control plane                                                                       |
+| deletecollection | DELETE /apis/rbac.authorization.k8s.io/v1/clusterroles      | all                         | not support                                                                                                                     |
+
+### Implementation plan
+
+To be supplemented.
```

Comment:
For the code location, it's better to:
1. fork kubernetes/kubernetes to karmada-io/kubernetes, and add only contain the necessary inline changes to make the kube-apiserver kube-apiserver programmable for implementing the fleet-apiserver.
2. The rest of the fleet-apiserver code should be stored in a separate place, better a staging repo in karmada-io/karmada, but a stand-alone repo can be an option as well.